### ISSUE TITLE
May you add this api that is SSL_get_negotiated_group in openssl-1.1.1f?

### ISSUE THREAD

**Issue by yangyangtiantianlonglong** [openssl/openssl#13295]
```javascript
diff --git a/openssl-1.1.1f/include/openssl/ssl.h b/openssl-1.1.1f/include/openssl/ssl.h
index ddf1443..94c3413 100644
--- a/open_source/openssl/src/openssl-1.1.1f/include/openssl/ssl.h
+++ b/open_source/openssl/src/openssl-1.1.1f/include/openssl/ssl.h
@@ -1305,6 +1305,7 @@ DECLARE_PEM_rw(SSL_SESSION, SSL_SESSION)
 # define SSL_CTRL_GET_MAX_PROTO_VERSION          131
 # define SSL_CTRL_GET_SIGNATURE_NID              132
 # define SSL_CTRL_GET_TMP_KEY                    133
+# define SSL_CTRL_GET_NEGOTIATED_GROUP           134
 # define SSL_CERT_SET_FIRST                      1
 # define SSL_CERT_SET_NEXT                       2
 # define SSL_CERT_SET_SERVER                     3
@@ -1402,6 +1403,8 @@ DECLARE_PEM_rw(SSL_SESSION, SSL_SESSION)
         SSL_ctrl(s,SSL_CTRL_SET_GROUPS_LIST,0,(char *)(str))
 # define SSL_get_shared_group(s, n) \
         SSL_ctrl(s,SSL_CTRL_GET_SHARED_GROUP,n,NULL)
+# define SSL_get_negotiated_group(s) \
+        SSL_ctrl(s,SSL_CTRL_GET_NEGOTIATED_GROUP,0,NULL)
 # define SSL_CTX_set1_sigalgs(ctx, slist, slistlen) \
         SSL_CTX_ctrl(ctx,SSL_CTRL_SET_SIGALGS,slistlen,(int *)(slist))
 # define SSL_CTX_set1_sigalgs_list(ctx, s) \
diff --git a/openssl-1.1.1f/ssl/s3_lib.c b/openssl-1.1.1f/ssl/s3_lib.c
index a987604..45be016 100644
--- a/open_source/openssl/src/openssl-1.1.1f/ssl/s3_lib.c
+++ b/open_source/openssl/src/openssl-1.1.1f/ssl/s3_lib.c
@@ -3639,6 +3639,11 @@ long ssl3_ctrl(SSL *s, int cmd, long larg, void *parg)
             }
             return id;
         }
+    case SSL_CTRL_GET_NEGOTIATED_GROUP:
+        {
+            const TLS_GROUP_INFO *inf = tls1_group_id_lookup(s->s3->group_id);
+            return inf == NULL ? 0 : inf->nid;
+        }
 #endif
     case SSL_CTRL_SET_SIGALGS:
         return tls1_set_sigalgs(s->cert, parg, larg, 0);
```

**Comment by mattcaswell** [comment]
We would not add such an API to 1.1.1 since this would be considered a new feature and we do not add new features to stable branches. You might want to consider creating a PR against master instead.

**Comment by yangyangtiantianlonglong** [comment]
https://github.com/openssl/openssl/blob/8ea761bf40e6578ecd95ec47772ef86a2e4d4607/include/openssl/ssl.h.in#L1440
https://github.com/openssl/openssl/blob/8ea761bf40e6578ecd95ec47772ef86a2e4d4607/ssl/s3_lib.c#L3675-L3680

The master already has  exist.

**Comment by mattcaswell** [comment]
> Only the bug can be merged before openssl-3.0.0 ？

Yes - we only allow bug and security fixes in 1.1.1.
